### DISCUSSION TITLE
A way to remove Anti Ailasing?

### DISCUSSION THREAD

**Discussion by jcsix694** [gonetz/GLideN64#2472] [Q&A]
Hey! Just wondering if theres a way to turn this off? I have MSAA set to off (0x) But there is still a slight antialising going on.

When I go into Frame Buffer and disable 'Emulate Frame Buffer', I get the look I am after. Do you have any suggestions?

![image](https://user-images.githubusercontent.com/59939332/112382421-bcec4e00-8ce3-11eb-8e63-131b031e9415.png)

I've only just started to use GlideN64, I used to use the default pluging. I only need this to take screenshots and cut out the backgroud, but with GlideN64 theres some Ani Aliasing going on which adds it on the model (And it looks really strange when the background is removed).

**Comment by Jj0YzL5nvJ** [comment]
`--set Video-GLideN64[MultiSampling]=0 --set Video-GLideN64[FXAA]=False --set Video-GLideN64[bilinearMode]=False --set Video-GLideN64[EnableHybridFilter]=False`

**Comment by gonetz** [comment]
It is not AA. The rendering image is seldom the same as the output one, because Video Interface (VI) usually scales rendering image before sending it to TV. Scaling adds this blurring. VI is not emulated when frame buffer emulation is off.
Hybrid filter is a way to reduce the artifacts of scaling, so I don't why @Jj0YzL5nvJ advises to disable it. 
bilinearMode option switches between N64 3point bilinear filtering and 4point standard one. Setting it to False just enables N64 3point bilinear.

**Reply by Jj0YzL5nvJ** [reply]
> Hybrid filter is a way to reduce the artifacts of scaling, so I don't why @Jj0YzL5nvJ advises to disable it.

To avoid "nearest-neighbor interpolation" with the background... hybrid filter can generate artifacts with black and red colours.
> I only need this to take screenshots and cut out the backgroud

**Reply by jcsix694** [reply]
> It is not AA. The rendering image is seldom the same as the output one, because Video Interface (VI) usually scales rendering image before sending it to TV. Scaling adds this blurring. VI is not emulated when frame buffer emulation is off.
> Hybrid filter is a way to reduce the artifacts of scaling, so I don't why @Jj0YzL5nvJ advises to disable it.
> bilinearMode option switches between N64 3point bilinear filtering and 4point standard one. Setting it to False just enables N64 3point bilinear.

Hybrid filter? Where is this?

**Comment by jcsix694** [comment]
Sorry it's been awhile!

@Jj0YzL5nvJ where would I put what you sent over? I've tried messing with the settings and can seem to remove it unless I untick emulate framebuffer, which different graphical bugs occur

**Reply by Jj0YzL5nvJ** [reply]
They are execution parameters for Mupen64Plus... example: https://imgur.com/a/LItmOz5

**Reply by jcsix694** [reply]
@Jj0YzL5nvJ Ah right! Im currently using PJ64 2.3.2 so I wouldnt of known about this. Any way to do that on that emulator too?

Sorry for the questions!

> version=26
> video\fullscreenWidth=1920
> video\fullscreenHeight=1080
> video\windowedWidth=1280
> video\windowedHeight=960
> video\fullscreenRefresh=60
> video\multisampling=0
> video\fxaa=0
> video\verticalSync=0
> texture\maxAnisotropy=0
> texture\bilinearMode=1
> texture\enableHalosRemoval=0
> texture\screenShotFormat=0
> generalEmulation\enableNoise=1
> generalEmulation\enableLOD=1
> generalEmulation\enableHWLighting=0
> generalEmulation\enableShadersStorage=1
> generalEmulation\enableCustomSettings=1
> graphics2D\correctTexrectCoords=0
> graphics2D\enableNativeResTexrects=0
> graphics2D\bgMode=1
> frameBufferEmulation\enable=1
> frameBufferEmulation\aspect=1
> frameBufferEmulation\nativeResFactor=0
> frameBufferEmulation\bufferSwapMode=0
> frameBufferEmulation\N64DepthCompare=0
> frameBufferEmulation\forceDepthBufferClear=0
> frameBufferEmulation\copyAuxToRDRAM=0
> frameBufferEmulation\copyFromRDRAM=0
> frameBufferEmulation\copyToRDRAM=2
> frameBufferEmulation\copyDepthToRDRAM=2
> frameBufferEmulation\fbInfoDisabled=0
> frameBufferEmulation\fbInfoReadColorChunk=1
> frameBufferEmulation\fbInfoReadDepthChunk=1
> frameBufferEmulation\enableOverscan=0
> frameBufferEmulation\overscanPalLeft=0
> frameBufferEmulation\overscanPalRight=0
> frameBufferEmulation\overscanPalTop=0
> frameBufferEmulation\overscanPalBottom=0
> frameBufferEmulation\overscanNtscLeft=0
> frameBufferEmulation\overscanNtscRight=0
> frameBufferEmulation\overscanNtscTop=0
> frameBufferEmulation\overscanNtscBottom=0
> textureFilter\txFilterMode=0
> textureFilter\txEnhancementMode=0
> textureFilter\txDeposterize=0
> textureFilter\txFilterIgnoreBG=0
> textureFilter\txCacheSize=104857600
> textureFilter\txHiresEnable=1
> textureFilter\txHiresFullAlphaChannel=1
> textureFilter\txHresAltCRC=0
> textureFilter\txDump=1
> textureFilter\txForce16bpp=0
> textureFilter\txCacheCompression=0
> textureFilter\txSaveCache=0
> textureFilter\txPath=C:/Users/jcsix/Desktop/Games/Nintendo 64/Project64/AKI Pro Wrestling/Plugin/GFX/hires_texture
> textureFilter\txCachePath=C:/Users/jcsix/Desktop/Games/Nintendo 64/Project64/AKI Pro Wrestling/Plugin/GFX/cache
> textureFilter\txDumpPath=C:/Users/jcsix/Desktop/Games/Nintendo 64/Project64/AKI Pro Wrestling/Plugin/GFX/texture_dump
> font\name=arial.ttf
> font\size=18
> font\color=@Variant(\0\0\0\x43\x1\xff\xff\xb5\xb5\xe6\xe6\x1d\x1d\0\0)
> gammaCorrection\force=0
> gammaCorrection\level=@Variant(\0\0\0\x87@\0\0\0)
> onScreenDisplay\showFPS=0
> onScreenDisplay\showVIS=0
> onScreenDisplay\showPercent=0
> onScreenDisplay\showInternalResolution=0
> onScreenDisplay\showRenderingResolution=0
> onScreenDisplay\osdPos=8
> debug\dumpMode=0
>

**Reply by oddMLan** [reply]
First make sure you're using the latest version of GLideN64 from the [Releases](https://github.com/gonetz/GLideN64/releases) section
Either Project64-Qt (x86) or Project64-WTL (x86) are fine. Qt has more features atm.
in `GLideN64.ini` set `generalEmulation\enableHybridFilter=1` to 0 and save.

**Reply by jcsix694** [reply]
@oddMLan Just done this and still getting the same results. Is there anything midding??

```
version=29
video\fullscreenWidth=1920
video\fullscreenHeight=1080
video\windowedWidth=1280
video\windowedHeight=960
video\fullscreenRefresh=60
video\multisampling=0
video\maxMultiSampling=16
video\fxaa=0
video\verticalSync=0
video\threadedVideo=0
texture\anisotropy=0
texture\maxAnisotropy=16
texture\bilinearMode=0
texture\enableHalosRemoval=0
generalEmulation\enableDitheringPattern=0
generalEmulation\enableDitheringQuantization=1
generalEmulation\enableHiresNoiseDithering=0
generalEmulation\rdramImageDitheringMode=3
generalEmulation\enableLOD=1
generalEmulation\enableHWLighting=0
generalEmulation\enableCoverage=0
generalEmulation\enableShadersStorage=1
generalEmulation\enableLegacyBlending=0
generalEmulation\enableHybridFilter=0
generalEmulation\enableFragmentDepthWrite=1
generalEmulation\enableCustomSettings=1
graphics2D\correctTexrectCoords=0
graphics2D\enableNativeResTexrects=0
graphics2D\bgMode=1
graphics2D\enableTexCoordBounds=0
frameBufferEmulation\enable=1
frameBufferEmulation\aspect=1
frameBufferEmulation\nativeResFactor=0
frameBufferEmulation\bufferSwapMode=0
frameBufferEmulation\N64DepthCompare=0
frameBufferEmulation\forceDepthBufferClear=0
frameBufferEmulation\copyAuxToRDRAM=0
frameBufferEmulation\copyFromRDRAM=0
frameBufferEmulation\copyToRDRAM=2
frameBufferEmulation\copyDepthToRDRAM=2
frameBufferEmulation\fbInfoDisabled=0
frameBufferEmulation\fbInfoReadColorChunk=0
frameBufferEmulation\fbInfoReadDepthChunk=1
frameBufferEmulation\copyDepthToMainDepthBuffer=0
frameBufferEmulation\enableOverscan=0
frameBufferEmulation\overscanPalLeft=0
frameBufferEmulation\overscanPalRight=0
frameBufferEmulation\overscanPalTop=0
frameBufferEmulation\overscanPalBottom=0
frameBufferEmulation\overscanNtscLeft=0
frameBufferEmulation\overscanNtscRight=0
frameBufferEmulation\overscanNtscTop=0
frameBufferEmulation\overscanNtscBottom=0
textureFilter\txFilterMode=0
textureFilter\txEnhancementMode=0
textureFilter\txDeposterize=0
textureFilter\txFilterIgnoreBG=0
textureFilter\txCacheSize=104857600
textureFilter\txHiresEnable=1
textureFilter\txHiresFullAlphaChannel=1
textureFilter\txHresAltCRC=0
textureFilter\txForce16bpp=0
textureFilter\txCacheCompression=0
textureFilter\txSaveCache=0
textureFilter\txEnhancedTextureFileStorage=0
textureFilter\txHiresTextureFileStorage=0
textureFilter\txNoTextureFileStorage=0
textureFilter\txPath=C:/Users/jcsix/Desktop/Games/Nintendo 64/Project64/Project64 2.3.2/Plugin/GFX/hires_texture
textureFilter\txCachePath=C:/Users/jcsix/Desktop/Games/Nintendo 64/Project64/Project64 2.3.2/Plugin/GFX/cache
textureFilter\txDumpPath=C:/Users/jcsix/Desktop/Games/Nintendo 64/Project64/Project64 2.3.2/Plugin/GFX/texture_dump
font\name=arial.ttf
font\size=18
font\color=@Variant(\0\0\0\x43\x1\xff\xff\xb5\xb5\xe6\xe6\x1d\x1d\0\0)
gammaCorrection\force=0
gammaCorrection\level=@Variant(\0\0\0\x87@\0\0\0)
onScreenDisplay\showFPS=0
onScreenDisplay\showVIS=0
onScreenDisplay\showPercent=0
onScreenDisplay\showInternalResolution=0
onScreenDisplay\showRenderingResolution=0
onScreenDisplay\showStatistics=0
onScreenDisplay\osdPos=8
hotkeys\hkTexDump=7
hotkeys\hkTexDumpEnabled=7
hotkeys\hkHdTexReload=0
hotkeys\hkHdTexReloadEnabled=0
hotkeys\hkHdTexToggle=0
hotkeys\hkHdTexToggleEnabled=0
hotkeys\hkTexCoordBounds=0
hotkeys\hkTexCoordBoundsEnabled=0
hotkeys\hkNativeResTexrects=0
hotkeys\hkNativeResTexrectsEnabled=0
hotkeys\hkVsync=0
hotkeys\hkVsyncEnabled=0
hotkeys\hkFBEmulation=0
hotkeys\hkFBEmulationEnabled=0
hotkeys\hkN64DepthCompare=0
hotkeys\hkN64DepthCompareEnabled=0
hotkeys\hkOsdVis=0
hotkeys\hkOsdVisEnabled=0
hotkeys\hkOsdFps=0
hotkeys\hkOsdFpsEnabled=0
hotkeys\hkOsdPercent=0
hotkeys\hkOsdPercentEnabled=0
hotkeys\hkOsdInternalResolution=0
hotkeys\hkOsdInternalResolutionEnabled=0
hotkeys\hkOsdRenderingResolution=0
hotkeys\hkOsdRenderingResolutionEnabled=0
hotkeys\hkForceGammaCorrection=0
hotkeys\hkForceGammaCorrectionEnabled=0
debug\dumpMode=0
```

**Reply by oddMLan** [reply]
> @oddMLan Just done this and still getting the same results. Is there anything midding??

Then that means the game resizes the screen by itself. Sorry. Disabling Framebuffer gets rid of the AA because it disables all Framebuffer operations, including those that resize the screen.

**Reply by Jj0YzL5nvJ** [reply]
ata4/angrylion-rdp-plus#75
